src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/api/UpsertAllocationRequest.kt

```diff
@@ -19,9 +19,8 @@ data class UpsertAllocationRequest(
   val bookingId: Long,
 
   @Schema(description = "The prisoner's pay band", example = "2")
-  @field:NotBlank
   @field:Length(min = 1, max = 12)
-  val payBandCode: String,
+  val payBandCode: String? = null,
```

Comment:
The main change - allocation pay band code can be null meaning there is no active pay band. This is to handle allocations to unpaid activities in DPS.